### ISSUE TITLE
Unified formatting for microformats and JSON-LD

### ISSUE THREAD

**Issue by Phyks** [scrapinghub/extruct#67]
Hi,

I am using `extruct` to extract metadata from emails, either in microformats or JSON+LD format. A very good point for this library is that in a single call one can extract all possible information from the message, that's super convenient!

However, I realized that the structure of the data returned from JSON+LD and microformats is quite different. For instance, microformats will return something like

```
{
  "type": "SOME_SCHEMA_URL",
  "properties": { /* A dict of properties */ }
}
```

whereas JSON+LD parsing would return something like

```
{
  "@type": "SOME_SCHEMA_URL",
  /* All the properties in keys here */
}
```

This is not so convenient as it implies that microformats and JSON+LD data should be handled in a different way, although they match the same schema.org schema.

Not sure if it is fine for `extruct` or if it should lie in another lib, but what about offering a way to have a standard representation of the extracted data. This could either be building a class object (a `struct` basically) from the fetched data for each type, or converting one of the format to the other. Not sure if this could already be offloaded to some external lib, but I could not find any doing the job so far.

Thanks!

**Comment by Kebniss** [comment]
Hi @Phyks, thank you for your feedback! This is an issue we will start to work on soon. Likely we'll implement another library that normalizes outputs from all syntaxes to the same standard. I'll make sure to ping you when we'll release it :-)

**Comment by Phyks** [comment]
Hi,

Oh, that's super cool news! Do you have any implementation ideas so far? Typically, I know about [Weboob](http://weboob.org/) (a scraping tool, in Python as well) and they have this implementation of `capabilities` which represents an analog of `schema.org` schemas. They are implemented as Python class with typed fields. For instance, an housing post might be represented by this class to standardize the data extracted from many websites: https://git.weboob.org/weboob/devel/blob/master/weboob/capabilities/housing.py#L75. I was thinking about doing a bit the same for `schema.org` schemas.

Let me know if you have any other implementation ideas and whenever this new library would become available, I would be greatly interested to try it out and contribute!

**Comment by lopuhin** [comment]
I think this was implemented in #70, also see follow-up #74

**Comment by Phyks** [comment]
Indeed, this will be solved after #70 and #74 are merged! Do you plan on releasing a new pip version afterwards? Thanks!

**Comment by lopuhin** [comment]
yes, we plan to make a release soon, I created 0.4 milestone today: https://github.com/scrapinghub/extruct/milestone/1

**Comment by Kebniss** [comment]
@Phyks, we released the new package version today :-)
